From 2895a1ffeb8c01c4fdb84cf4268cd968a743e332 Mon Sep 17 00:00:00 2001
From: Stuart Yoder <stuart.yoder@arm.com>
Date: Thu, 5 Aug 2021 17:03:45 -0500
Subject: [PATCH 23/37] uefi-sct/SctPkg: Secure Boot: fix image generation for
 TestImage10

For the test assertion covered by TestImage10 we need both a signed
image to use for the test, plus an unsigned image to use to create
a hash used for dbx revocation.

Because of how CommonGenFramework.sh works it overwrote the signed
image in the target image with the unsigned image.

Fix this by renaming the unsigned image so it will not be copied
as a dependency file by CommonGenFramework.sh.

Signed-off-by: Stuart Yoder <stuart.yoder@arm.com>
---
 .../SecureBoot/BlackBoxTest/Dependency/Keys/GNUmakefile       | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Keys/GNUmakefile b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Keys/GNUmakefile
index 8942221d..00bc1e48 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Keys/GNUmakefile
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Dependency/Keys/GNUmakefile
@@ -85,7 +85,7 @@ TestImage9.bin: RevokedCert4
 	sbsign --key $(TARGET)_RevokedCert4.key --cert $(TARGET)_RevokedCert4.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
 
 TestImage10.bin: SignCert3
-	cp $(SOURCE_DIR)/Application3.efi $(TARGET)_Unsigned_$(@)
+	cp $(SOURCE_DIR)/Application3.efi $(TARGET)_Unsigned_$(@).tmp
 	cp $(SOURCE_DIR)/Application3.efi $(TARGET)_$(@)
 	sbsign --key $(TARGET)_SignCert3.key --cert $(TARGET)_SignCert3.crt $(TARGET)_$(@) --output $(TARGET)_$(@)
 
@@ -195,7 +195,7 @@ DBXRevokedList1.auth: RevokedCert1 RevokedCert2 RevokedCert3 RevokedCert4
 	cert-to-efi-sig-list $(BIN_DIR)/SecureBoot_RevokedCert2.crt $(BIN_DIR)/SecureBoot_RevokedCert2.esl
 	cert-to-efi-sig-list $(BIN_DIR)/SecureBoot_RevokedCert3.crt $(BIN_DIR)/SecureBoot_RevokedCert3.esl
 	cert-to-efi-sig-list $(BIN_DIR)/SecureBoot_RevokedCert4.crt $(BIN_DIR)/SecureBoot_RevokedCert4.esl
-	hash-to-efi-sig-list $(BIN_DIR)/SecureBoot_Unsigned_TestImage10.bin $(BIN_DIR)/SecureBoot_RevokedHash1.esl
+	hash-to-efi-sig-list $(BIN_DIR)/SecureBoot_Unsigned_TestImage10.bin.tmp $(BIN_DIR)/SecureBoot_RevokedHash1.esl
 	cat $(BIN_DIR)/SecureBoot_RevokedCert1.esl $(BIN_DIR)/SecureBoot_RevokedCert2.esl $(BIN_DIR)/SecureBoot_RevokedCert3.esl  $(BIN_DIR)/SecureBoot_RevokedCert4.esl $(BIN_DIR)/SecureBoot_RevokedHash1.esl > $(BIN_DIR)/SecureBoot_DBXRevokedList1.esl
 	sign-efi-sig-list -c $(TEST_KEK1_CRT) -k $(TEST_KEK1_KEY) -t "$(FUTURE_DATE2)" dbx $(BIN_DIR)/SecureBoot_DBXRevokedList1.esl $(BIN_DIR)/SecureBoot_DBXRevokedList1.auth
 
-- 
2.17.1

